### ISSUE TITLE
Centered Height With Flexbox

### ISSUE THREAD

**Issue by xcrap** [metafizzy/flickity#665]
<img width="1475" alt="screen shot 2017-12-08 at 04 44 35" src="https://user-images.githubusercontent.com/828596/33752661-9e1c6890-dbd2-11e7-8636-92d4dc5c3f9d.png">

I'm trying to develop a Carousel the same way I had in Masterslider. 
On top I'm using flickity and on button I'm using Masterslider.

The problem is centering the images to create a transform height transition. 
I've tried to add flexbox to carousel-items and then vertically center the image, the problem is that it does not work without defining width and I'm trying to find a solution where I could add images with different widths but I really wanted that transform effect.

A live demo is available here:
https://www.boscana.de/collections/embrace-kollektion/

If I make for example:
.carousel-cell img { transform:scale(.9) }
.carousel-cell.is-selected img { transform:scale(1) }
There will be a padding on the smaller images, what i don't pretend.
The same will happen with vertical margins and paddings.

If I use
.carousel-cell img { max-height:80% }
.carousel-cell.is-selected img { max-height:100%; }
This is close but the selected image width will be shorter than the others and will cause an overlap of the selected image.
 
Anyone have any idea on how to achieve that below carousel effect that would work with variable width images ? I've tried with vertical paddings, margins, flexbox, image transform scale but it's not the same.

**Comment by yaodingyd** [comment]
Changing height is not a problem. Your issue is to change cell width when selecting cell. IMO this is not possible with Flickity.

Flickity first calculates cell width to position cells. So before `previous`/`next` is called, cells' width should already be determined. Even if cells' width is changed during selecting cell, the new width would not effect cells' position, thus the overlap of selected cell.

I have one imperfect [demo](https://codepen.io/earthmount/pen/rdeMYM) here that partially meets your requirement:
1. As it's not possible to change width during selecting, I add a custom selected class to selected cell when `settle`.
2. With transition effect, cell widths is constantly changing, so cells need to be `reposition`ed too. Put `reposition` in `requestAnimationFrame` to achieve 60FPS.
3. When cell width is transitioning, prev/next is disabled as it would produce sudden cell shifts.

Basically, constantly changing cell width requires a lot of `reposition` and it will lower your performance. It is not a good practice for Flickity.
